library/kani/src/lib.rs

```diff
@@ -316,6 +107,24 @@ macro_rules! cover {
     };
 }
 
+impl<T> Arbitrary for std::boxed::Box<T>
+where
+    T: Arbitrary,
+{
```

Comment:
Don't we want these implementations to live in `library/kani_core/src/arbitrary.rs`?

Reply:
I would say no for now. A key requirement for anything in Kani core is that it needs to be something that has an implementation in the core library as well as std, this allows us to switch between the two source locations at compile time once the macros get expanded. Based on what i found, it doesnt seem like there are no_core implementations of boxed or box or time. 

What this means is that, we can have some methods as extensions of our core API, but are not a part of our core API since `kani_core` is a minimal dependency crate.

Reply:
But shouldn't this be inside `kani/src/arbitrary.rs`